File: Prefix_Sum/Pattern4/Pattern4.py
nums=[1,0,-1,4,2,-2,-4]
dct={}
dct[0]=1
sum=0
count=0

for i in range(len(nums)):
    sum+=nums[i]

    if dct.get(sum):
        count+=dct.get(sum)
        dct[sum]=dct.get(sum)+1
    else:
        dct[sum]=1
print(count)

File: Prefix_Sum/Pattern4/Pattern4.java
import java.util.*; 
public class Pattern4 {
    public static void main(String[] args) {
        int[] nums={1,0,-1,4,2,-2,-4};
        Map<Integer,Integer> map=new HashMap<>();
        map.put(0,1);
        int sum=0;
        int count=0;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            if (map.containsKey(sum)){
                count+=map.get(sum);
                map.put(sum,map.get(sum)+1);
            }
            else{
                map.put(sum,1);
            }
        }
        System.out.println(count);
    }    
}
